x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ingest/SetSecurityUserProcessor.java

```diff
@@ -34,20 +35,29 @@
 
     public static final String TYPE = "set_security_user";
 
+    private final Logger logger = LogManager.getLogger();
+
     private final SecurityContext securityContext;
     private final String field;
     private final Set<Property> properties;
 
     public
     SetSecurityUserProcessor(String tag, SecurityContext securityContext, String field, Set<Property> properties) {
         super(tag);
-        this.securityContext = Objects.requireNonNull(securityContext, "security context must be provided");
+        this.securityContext = securityContext;
+        if (this.securityContext == null) {
+            logger.warn("Creating processor [{}] (tag [{}]) on field [{}] but no security context is available" +
+                " - this processor will fail at runtime if it is used", TYPE, tag, field);
```

Comment:
Not sure how feasible or necessary it is to test the warning message. Another nitpick is maybe change "no security context is available" to something like "security is not enabled" for consistency with other exception message and easier to understand for users.

Reply:
I'll come up with something. My concern is trying to diagnose a root cause from secondary symptoms (I know there's no security context, I don't _know_ for sure exactly why that is), but I'll sort something out.